invoice/src/main/java/org/killbill/billing/invoice/InvoiceDispatcher.java

```diff
@@ -348,15 +349,11 @@ private Invoice processAccountInternal(final boolean parkedAccount,
         try {
             // Make sure to first set the BCD if needed then get the account object (to have the BCD set)
             final BillingEventSet billingEvents = billingApi.getBillingEventsForAccountAndUpdateAccountBCD(accountId, dryRunArguments, context);
-            if (billingEvents.isAccountAutoInvoiceOff()) {
+            if (!isApiCall && billingEvents.isAccountAutoInvoiceOff()) {
                 return null;
             }
 
-            // Avoid pulling all invoices when AUTO_INVOICING_OFF is set since we will disable invoicing later
-            // (Note that we can't return right away as we send a NullInvoice event)
```

Comment:
It looks like we were returning right away before. Was this a stale comment? 🤔

Reply:
Yep, I am also having trouble with the comment, as it does not make sense considering [we were returning right away on the line before](https://github.com/killbill/killbill/blob/killbill-0.22.25/invoice/src/main/java/org/killbill/billing/invoice/InvoiceDispatcher.java#L352) when `AUTO_INVOICING_OFF` was set.